frontend/src/components/widgets/CameraInput/CameraInput.tsx

```diff
@@ -0,0 +1,579 @@
+  /**
+   * Base64-encoded image data of the current frame from the camera.
+   */
+  imgSrc: string | null
+
+  shutter: boolean
+
+  minShutterEffectPassed: boolean
+  /**
+   * List of files dropped on the FileUploader by the user. This list includes
+   * rejected files that will not be updated.
+   */
+  files: UploadFileInfo[]
+
+  /**
+   * The most recent file ID we've received from the server. This gets sent
+   * back to the server during widget update so that it clean up
+   * orphaned files. File IDs start at 1 and only ever increase, so a
+   * file with a higher ID is guaranteed to be newer than one with a lower ID.
+   */
+  newestServerFileId: number
+  /** Represents whether the component is in clear photo mode
+   * Time interval between `Clear Photo` button clicked and access to Webcam recived again
+   */
+  inClearPhotoMode: boolean
+}
+
+class CameraInput extends React.PureComponent<Props, State> {
+  private localFileIdCounter = 1
+
+  private RESTORED_FROM_WIDGET_STRING = "RESTORED_FROM_WIDGET"
+
+  private readonly formClearHelper = new FormClearHelper()
+
+  public constructor(props: Props) {
+    super(props)
+    this.state = this.initialValue
+  }
+
+  private getProgress = (): number | null | undefined => {
+    if (
+      this.state.files.length > 0 &&
+      this.state.files[this.state.files.length - 1].status.type === "uploading"
+    ) {
+      const status = this.state.files[this.state.files.length - 1]
+        .status as UploadingStatus
+      return status.progress
+    }
+    return undefined
+  }
+
+  private setInClearPhotoMode = (inClearPhotoMode: boolean): void => {
+    this.setState({ inClearPhotoMode })
+  }
+
+  private handleCapture = (imageSrc: string | null): void => {
+    if (imageSrc === null) {
+      return
+    }
+
+    this.setState({
+      imgSrc: imageSrc,
+      shutter: true,
+      minShutterEffectPassed: false,
+    })
+
+    urltoFile(imageSrc, `camera-input-${new Date().toISOString()}.jpg`)
+      .then(file => this.uploadFile(file))
+      .catch(err => {
+        logError(err)
+      })
+
+    setTimeout(() => {
+      this.setState({
+        imgSrc: imageSrc,
+        shutter: this.state.shutter,
+        minShutterEffectPassed: true,
+      })
+    }, 150)
+  }
+
+  private removeCapture = (): void => {
+    if (this.state.files.length === 0) {
+      return
+    }
+
+    this.state.files.forEach(file => this.deleteFile(file.id))
+
+    this.setState({
+      imgSrc: null,
+      inClearPhotoMode: true,
+    })
+  }
+
+  get initialValue(): State {
+    const emptyState = {
+      files: [],
+      newestServerFileId: 0,
+      imgSrc: null,
+      // Represents whether file uploading is done
+      shutter: false,
+      // Represents whether minimum shutter time has passed
+      minShutterEffectPassed: true,
+      // Represents whether the component is in clear photo mode
+      // Time interval between `Clear Photo` button clicked and access to Webcam recived again
```

Comment:
I think these comments should probably be moved into the `State` interface rather than have them live here.

I'd actually argue that most of the comments beginning with "Represents..." should probably be removed entirely as they don't add much more than the variable names alone.

Reply:
@kajarenc , I have removed the comments in my unit_tests pr. please review it. https://github.com/kajarenc/streamlit/pull/7